core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -319,6 +319,39 @@ impl EthNamespace {
         Ok(tx_count?.map(|(_, count)| count))
     }
 
+    pub async fn get_block_receipts_impl(
+        &self,
+        block_id: BlockId,
+    ) -> Result<Vec<TransactionReceipt>, Web3Error> {
+        const METHOD_NAME: &str = "get_block_receipts";
+
+        let method_latency = API_METRICS.start_block_call(METHOD_NAME, block_id);
+
+        let transaction_count = self
+            .get_block_transaction_count_impl(block_id)
+            .await?
+            .ok_or(Web3Error::NoBlock)?;
+
+        let mut receipts = Vec::with_capacity(transaction_count.as_usize());
+
+        for index in 0..transaction_count.as_usize() {
+            let transaction = self
+                .get_transaction_impl(TransactionId::Block(block_id, index.into()))
+                .await?
+                .ok_or(Web3Error::InternalError)?;
+
+            receipts.push(
+                self.get_transaction_receipt_impl(transaction.hash)
+                    .await?
+                    .ok_or(Web3Error::InternalError)?,
+            );
+        }
+
+        method_latency.observe_without_diff();
```

Comment:
Since the query involves `block_id`, you *can* observe block diff using `report_latency_with_block_id()`, at least theoretically. One way is to get block number is to get it from a transaction receipt (not 100% reliable, since we have empty miniblocks); or it may appear as an output of some intermediate query (e.g., `get_block_by_web3_block_id()` as proposed above).